cloud/src/autogluon/cloud/predictor/cloud_predictor.py

```diff
@@ -292,71 +291,99 @@ def _prepare_data(self, data, filename, output_type="csv"):
         converter = FormatConverterFactory.get_converter(output_type)
         return converter.convert(data, path, filename)
 
-    def _upload_fit_image_artifact(self, images, bucket, key_prefix):
-        if images is not None:
-            if is_s3_url(images):
-                fileexts = (".tar.gz", ".bz2", ".zip")  # More extensions?
-                assert images.endswith(fileexts), "Please provide a s3 path to a compressed file"
-            else:
-                image_zip_filename = images
-                if os.path.isdir(images):
-                    image_zip_filename = os.path.basename(os.path.normpath(images))
-                    zipfolder(image_zip_filename, images)
-                    image_zip_filename += ".zip"
-                else:
-                    assert is_compressed_file(
-                        images
-                    ), "Please provide a compressed file or a folder containing the images"
-                logger.log(20, "Uploading images ...")
-                images = self.sagemaker_session.upload_data(
-                    path=image_zip_filename,
-                    bucket=bucket,
-                    key_prefix=key_prefix,
-                )
-                logger.log(20, "Images uploaded successfully")
-        return images
+    def _upload_fit_image_artifact(self, image_dir_path, bucket, key_prefix):
+        upload_image_path = None
+        if image_dir_path is not None:
+            image_zip_filename = image_dir_path
+            assert os.path.isdir(image_dir_path), "Please provide a folder containing the images"
+            image_zip_filename = os.path.basename(os.path.normpath(image_dir_path))
+            logger.log(20, "Zipping images ...")
+            zipfolder(image_zip_filename, image_dir_path)
+            image_zip_filename += ".zip"
+            logger.log(20, "Uploading images ...")
+            upload_image_path = self.sagemaker_session.upload_data(
+                path=image_zip_filename,
+                bucket=bucket,
+                key_prefix=key_prefix,
+            )
+            logger.log(20, "Images uploaded successfully")
+        return upload_image_path
+
+    def _find_common_path_and_replace_image_column(self, data, image_column):
+        common_path = os.path.commonpath(data[image_column].tolist())
+        common_path_head = os.path.split(common_path)[0]  # we keep the base dir to match zipping behavior
+        data[image_column] = [os.path.relpath(path, common_path_head) for path in data[image_column]]
```

Comment:
`data[image_column] = data[image_column].apply(lambda path: os.path.relpath(path, common_path_head))`

TIOLI: This is probably more efficient because it avoids going to a python list and back to pandas Series